### PR TITLE
initial basic express app

### PR DESCRIPTION
**Pull request by tavvy**
Based on https://github.com/shunterjs/example/tree/master/backend-spec

I have a question about the license we want to use. Any input?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+
+'use strict';
+
+var port = process.env.PORT || 3000;
```
Comment (rowanmanning) [inline app.js]: Because the example app currently defaults to `5000`, could we change this default to `5000` as well?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+var Venue = require('./models/venue');
+
+var app = express();
+
+/* 
+ * MIDDLEWARE
+ * serve images from the public directory
+ */
+app.use(express.static('./public/img/', {
+	setHeaders: function(res, path, stat) {
```
Comment (rowanmanning) [inline app.js]: I don't think you'll need the `setHeaders` config – as far as I know `express.static` detects it based on the file extension :)

File: app.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	next();
+});
+
+/* 
+ * HOME
+ * home page
+ */
+app.get('/', function(req, res, next){
+	Home.get(function(err, result) {
+		if(err) return next(err);
```
Comment (rowanmanning) [inline app.js]: Doesn't comply to our styleguide. Whitespace:

``` js
if (err) {
    return next(err);
}
```

File: models/home.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+/*
+ * GET
+ * return the homepage data
+ */
+exports.get = function(callback) {
+	var venues = null;
+
+	// mock request to fetch venues data
+	venues = JSON.parse(JSON.stringify(db));
```
Comment (rowanmanning) [inline models/home.js]: Don't think this is necessary, may as well just use the db directly as we're not making modifications to it

File: models/home.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+	if (!venues) return callback({error: 'Venues not found'});
```
Comment (rowanmanning) [inline models/home.js]: Should be an `Error` object (also conditional whitespace)

File: package.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  "main": "app.js",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/shunterjs/example-backend-node-express.git"
+  },
+  "author": "Shunterjs",
+  "contributors": [
+    "Adam Tavener (http://www.tavvy.co.uk/)"
+  ],
+  "license": "ISC",
```
Comment (rowanmanning) [inline package.json]: Could we go `LGPL-3.0` to match Shunter itself?

**Comment by rowanmanning** [comment]
Mostly small comments/changes. Looks great! Good use of middleware to set a default Content-Type – wouldn't have thought of that

File: .jscsrc [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	"requireSpacesInFunction": {
+		"beforeOpeningCurlyBrace": true
+	},
+	"validateIndentation": "\t",
+	"validateLineBreaks": "LF",
+	"validateNewlineAfterArrayElements": true,
+	"validateParameterSeparator": ", ",
+	"validateQuoteMarks": "'",
+
+	"excludeFiles": [
```
Comment (rowanmanning) [inline .jscsrc:87]: This array only needs to contain `node_modules` and `public` now.

**Comment by rowanmanning** [comment]
Looks awesome, just the one comment then feel free to merge!

**Comment by tavvy** [comment]
thanks :+1:

**Pull request merged**